client/components/segmentation-survey/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCallback } from 'react';
+import DocumentHead from 'calypso/components/data/document-head';
+import SurveyContainer from 'calypso/components/survey-container';
+import { Question } from 'calypso/components/survey-container/types';
+import {
+	useCachedAnswers,
+	useSaveAnswersMutation,
+	useSurveyStructureQuery,
+} from 'calypso/data/segmentaton-survey';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import useSegmentationSurveyNavigation from './hooks/use-segmentation-survey-navigation';
+
+type SegmentationSurveyProps = {
+	surveyKey: string;
+	onBack?: () => void; // This is a function that navigates to the previous step
+	onComplete?: () => void; // This is a function that navigates to the next step
+};
+
+const SegmentationSurvey = ( { surveyKey, onBack, onComplete }: SegmentationSurveyProps ) => {
+	const { data: questions } = useSurveyStructureQuery( { surveyKey } );
+	const { mutateAsync, isPending } = useSaveAnswersMutation( { surveyKey } );
+	const { answers, setAnswers, clearAnswers } = useCachedAnswers( surveyKey );
+
+	const onChangeAnswer = useCallback(
+		( questionKey: string, value: string[] ) => {
+			const newAnswers = { ...answers, [ questionKey ]: value };
+			setAnswers( newAnswers );
+		},
+		[ answers, setAnswers ]
+	);
+
+	const handleSave = useCallback(
+		async ( currentQuestion: Question, answerKeys: string[] ) => {
+			try {
+				await mutateAsync( {
+					questionKey: currentQuestion.key,
+					answerKeys,
+				} );
+
+				if ( questions?.[ questions.length - 1 ].key === currentQuestion.key ) {
+					clearAnswers();
+					onComplete?.();
+				}
+			} catch ( e ) {
+				const error = e as Error;
+
+				recordTracksEvent( 'calypso_segmentation_survey_error', {
+					survey_key: surveyKey,
+					question_key: currentQuestion.key,
+					answer_keys: answerKeys,
+					error_message: error.message,
+				} );
+			}
+		},
+		[ clearAnswers, mutateAsync, onComplete, questions, surveyKey ]
+	);
+
+	const onContinue = useCallback(
+		async ( currentQuestion: Question ) => {
+			await handleSave( currentQuestion, answers[ currentQuestion.key ] || [] );
+		},
+		[ answers, handleSave ]
+	);
+
+	const onSkip = useCallback(
+		async ( currentQuestion: Question ) => {
+			await handleSave( currentQuestion, [ 'skipp' ] );
```

Comment:
Small typo, but will result in skipping not working:

![Markup on 2024-04-30 at 17:30:33](https://github.com/Automattic/wp-calypso/assets/25105483/0aea3eaf-e588-444b-ba02-6053548b1cd3)

```suggestion
			await handleSave( currentQuestion, [ 'skip' ] );
```

Reply:
Indeed, I forgot to remove it, it was only for testing the error logs. Thank you for catching that Ivan!